### ISSUE TITLE
Malformed `payload` in webhook POST Crashes Server?

### ISSUE THREAD

**Issue by blaskovicz** [mattermost/platform#1050]
I downloaded the mattermost 1.0.0 tarball, configured it to use postgres, and then ran it with `./bin/platform`.

Upon making the following curl request (albeit malformed):

```
$ curl -XPOST -i -d'payload={"username":"'`whoami`'"text": "sup? from a webhook on '`hostname`' at '`date`'"}' -v http://<hostname>:<port>/hooks/<hook-auth-id>
```

...the server seems to crash with:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x5034f2]

goroutine 12208 [running]:
github.com/mattermost/platform/api.func·010()
        /home/travis/gopath/src/github.com/mattermost/platform/api/post.go:192 +0x41e2
created by github.com/mattermost/platform/api.fireAndForgetNotifications
        /home/travis/gopath/src/github.com/mattermost/platform/api/post.go:420 +0x124

goroutine 1 [chan receive, 46 minutes]:
main.main()
        /home/travis/gopath/src/github.com/mattermost/platform/mattermost.go:68 +0x6d1

goroutine 6900 [select]:
code.google.com/p/log4go.func·002()
        /home/travis/gopath/src/github.com/mattermost/platform/Godeps/_workspace/src/code.google.com/p/log4go/filelog.go:84 +0xa53
created by code.google.com/p/log4go.NewFileLogWriter
        /home/travis/gopath/src/github.com/mattermost/platform/Godeps/_workspace/src/code.google.com/p/log4go/filelog.go:116 +0x3da
```

**Comment by coreyhulen** [comment]
Thanks for the report.  I'm 99% sure this is the same issue as https://mattermost.atlassian.net/browse/PLT-414

**Comment by crspeller** [comment]
Hopefully fixed with #1142
